pkg/apis/core/v1/types.go

```diff
@@ -650,10 +650,15 @@ type (
 		// RelativeURL is the exposed URL for external client to access a function with.
 		RelativeURL string `json:"relativeurl"`
 
+		// Deprecated: Use Methods instead of Method.
 		// HTTP method to access a function.
 		// +optional
 		Method string `json:"method"`
 
+		// HTTP methods to access a function
+		// +optional
+		Methods []string `json:"methods"`
```

Comment:
Optional fields should have omit empty tags 
```suggestion
		Methods []string `json:"methods,omitempty"`
```

Reply:
This is intentional, just trying to avoid nil checks as well in future this would be a compulsory or non-nil field once we remove Method.

Reply:
I think we can remove the optional tag in the next API version and then it would be okay to not have the `omitempty` tag there but in this version as we're marking it as optional , it's recommended to have the tag. https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required